src/adapter/src/coord/timeline.rs

```diff
@@ -840,6 +840,12 @@ impl Coordinator {
         let mut id_bundle: CollectionIdBundle = self
             .index_oracle(compute_instance)
             .sufficient_collections(item_ids.iter());
+        // Queries may get auto-routed to the mz_introspection cluster, so we include indexes on
```

Comment:
What happens if they aren't auto routed. Does this over guess and do something wrong?

Reply:
There's some weird interactions with timedomains and clusters, I just started a discussion here: https://materializeinc.slack.com/archives/C02FWJ94HME/p1695214693135449

This change makes it so that indexes in BOTH the active cluster and the `mz_introspection` cluster are included in the timedomain. Previously, only indexes in the active cluster were included in the timedomain. Without this change, the issue is that if you're query is auto-routed to `mz_introspection` and hits an index, then it will be aborted because that index is outside of the timedomain.

> What happens if they aren't auto routed. Does this over guess and do something wrong?

If they aren't auto-routed, then we have the same behavior as before this change. The query will use the current cluster, which also has it's indexes in the timedomain (if we haven't switched clusters). The read holds on the indexes in `mz_introspection` end up being wasteful because we don't use them, but we don't do anything wrong.